File: Sorting/SelectionSort.java
import java.io.*;
import java.util.Scanner;

public class SelectionSort 
{
    static void sort(int []arr)
    {
        int n = arr.length;

        for(int i=0; i<n-1; i++)
        {
            int min = i;
            for(int j=i+1; j<n; j++)
            {
                if(arr[j] < arr[min])
                {
                    min = j;
                }
            }

            int temp = arr[min];
            arr[min] = arr[i];
            arr[i] = temp;
        }
    }

    static void display(int []arr)
    {
        for(int i=0; i<arr.length; i++)
        {
            System.out.print(arr[i] + " ");
        }
    }

    public static void main(String args[])
    throws IOException
    {
        Scanner sc = new Scanner(System.in);
        int arr[] = {};

        System.out.print("Enter size of array: ");
        int size = sc.nextInt(); 
        arr = new int[size];

        System.out.println("Enter the Elements");

        for(int i=0; i<size; i++)
        {
            System.out.print("Element " + i + " : ");
            arr[i] = sc.nextInt();
        }

        System.out.println("Array before Selection sort");
        display(arr);
        System.out.println();

        sort(arr);
    
        System.out.println("Array after Selection sort");
        display(arr);

        sc.close();
    }
}

File: Selectionsort.py

def selectionSort(array, size):
	
	for ind in range(size):
		min = ind

		for j in range(ind + 1, size):
			if array[j] < array[min]:
				min = j
		(array[ind], array[min]) = (array[min], array[ind])

arr = [9,6,8,4,21,3,1,6,10,0]
size = len(arr)
selectionSort(arr, size)
print('The array after sorting in Ascending Order by selection sort is:')
print(arr)
